tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -187,8 +248,67 @@ def _run(self):
                                         hostname not in network_ip):
                                         b_index += 1
                                         continue
+
+                                    sub_vol_size = (int(volumes['volume%s.brickcount' % index])) / int(
+                                        volumes[
+                                            'volume%s.subvol_count' % index
+                                        ]
+                                    )
+                                    brick_name = NS.node_context.fqdn + ":" + volumes[
+                                        'volume%s.brick%s.path' % (
+                                            index, b_index
+                                        )
+                                    ].split(":")[-1].replace("/","_")
+
+                                    # Raise alerts if the brick path changes
+                                    try:
+                                        stored_brick_status = NS._int.client.read(
+                                            "clusters/%s/Bricks/all/%s/status" % (
+                                                NS.tendrl_context.integration_id,
+                                                brick_name
+                                            )
+                                        ).value
+                                        current_status = volumes.get(
+                                            'volume%s.brick%s.status' % (
+                                                index,
+                                                b_index
+                                            )
+                                        )
+                                        if current_status != stored_brick_status:
+                                            msg = "Status of brick: %s under volume %s changed from %s to %s" % (
+                                                volumes[
+                                                    'volume%s.brick%s.path' % (
+                                                        index, b_index
+                                                    )
+                                                ],
+                                                volumes['volume%s.name' % index],
+                                                stored_brick_status,
+                                                current_status
+                                            )
+                                            self._emit_event(
+                                                volumes['volume%s.name' % index],
```

Comment:
for cases like brick where info about its volume and cluster both are important, please have an additional parameter under tags called the plugin_instance and have its value in accordance with https://github.com/Tendrl/node-monitoring/blob/develop/tendrl/node_monitoring/plugins/tendrl_glusterfs_brick_utilization.py#L286

Now, the reason for not having additional fields like vol_name, brick_path extra under the dict tags is , that collectd does not allow us to have custom additional fields in tags as the tags attribute for collectd generated alerts come directly from collectd based on how the plugin is configured(an example is the above link) and only a few reserved fields can be played around with which leaves plugin_instance as the best attribute choice left...
So if we decide to have custom fields in our(tendrl generated) alerts unless absolutely necessary will render the alerting module validations and alert clearing logic less generic...